execution/function/operator.go

```diff
@@ -341,24 +341,22 @@ func DropMetricName(l labels.Labels) (labels.Labels, labels.Label) {
 
 // dropLabel removes the label with name from l and returns the dropped label.
 func dropLabel(l labels.Labels, name string) (labels.Labels, labels.Label) {
-	if len(l) == 0 {
+	var ret labels.Label
+
+	if l.IsEmpty() {
 		return l, labels.Label{}
 	}
 
-	if len(l) == 1 {
-		if l[0].Name == name {
-			return l[:0], l[0]
+	b := labels.NewScratchBuilder(l.Len())
```

Comment:
What I meant was we could inject a `ScratchBuilder` as an argument in the function and call `Reset` here. This way we can reuse the same scratch builder at the call site.

Reply:
Gotcha...

You think initializing it with a `expected` size make sense? An other alternative would be receive and return a reference for the builder in the function (first nil) and create the builder inside of it (in case is nil) and return.

Like:

```
func DropMetricName(l labels.Labels, b *labels.ScratchBuilder) (labels.Labels, labels.Label, *labels.ScratchBuilder)
```

Reply:
I think it's fine for now. We could also explore an approach of having label builder pools in the future so we can reuse these objects/buffers to reduce allocs.

Reply:
Im more than happy to create a follow up PR! :D